falcon/falcon3-10B-trt-llm-spec-dec/config.yaml

```diff
@@ -0,0 +1,48 @@
+model_metadata:
+  tags:
+    - openai-compatible
+  example_model_input:
+    frequency_penalty: 1
+    max_tokens: 512
+    messages:
+      - content: You are a knowledgable, engaging, history teacher.
+        role: system
+      - content: What was the role of falcons in Lord of the Rings?
+        role: user
+    stream: true
+    temperature: 0.3
+    top_k: 40
+    top_p: 1.0
+  repo_id: tiiuae/Falcon3-10B-Instruct
+model_name: Falcon3-10B-Instruct-with-1B-SpecDec
+resources:
+  accelerator: H100
+  cpu: "1"
+  memory: 24Gi
+  use_gpu: true
+trt_llm:
+  build:
+    base_model: llama
+    checkpoint_repository:
+      repo: tiiuae/Falcon3-10B-Instruct
+      source: HF
+    max_seq_len: 8192
+    num_builder_gpus: 1
+    plugin_configuration:
+      paged_kv_cache: true
+      use_paged_context_fmha: true
+    quantization_type: no_quant
+    tensor_parallel_count: 1
+    speculator:
+      speculative_decoding_mode: DRAFT_TOKENS_EXTERNAL
+      checkpoint_repository:
+        # This is the checkpoint repository for the speculative decoding model.
+        # Models of the same architecture may be fine-tuned for higher performance
+        repo: tiiuae/Falcon3-1B-Instruct
+        source: HF
+      num_draft_tokens: 4
+  runtime:
+    enable_chunked_context: true
+    kv_cache_free_gpu_mem_fraction: 0.62
+    request_default_max_tokens: 1000
+    total_token_limit: 500000
```

Comment:
Should this be a round number?

Reply:
total_token_limit defines a briton setting for how many tokens are queued inside C++. If queueing too many requests, we overload runtime. Not important, as long as it's in a decent range. 

total_token_limit = 500000 is default.

kv_cache_free_gpu_mem_fraction: 0.62 is a hard setting. 
Falcon-3-10B needs around 20GB Vram for weights. We have 20GB Vram left. 
Of the rest of the 20GB we allocate 62% to Falcon-10B. The rest of 7.6GB go to the 1B model. 2 GB out of the 7.6Gb will go to the weights, while 5.6GB are for activations and KV cache.